src/__tests__/components/fields/location-field/location-field.spec.tsx

```diff
@@ -644,6 +683,69 @@ describe("location-input-group", () => {
 				expect(screen.queryByTestId(TestHelper.generateId(COMPONENT_ID, "modal", "show"))).toBeVisible();
 			});
 		});
+
+		//Confirm Location Events
+		describe("Confirm Location events", () => {
+			beforeEach(async () => {
+				getCurrentLocationSpy.mockRejectedValue({
+					code: 1,
+				});
+
+				renderComponent({
+					withEvents: true,
+					locationDetails: {
+						errors: {
+							generic: "throw something",
+						},
+					},
+				});
+
+				await waitFor(() => window.dispatchEvent(new Event("online")));
+
+				getLocationInput().focus();
+
+				await waitFor(() => {
+					expect(getCurrentLocationErrorModal(true)).toBeInTheDocument();
+				});
+
+				within(getCurrentLocationErrorModal(true)).getByRole("button").click();
+			});
+
+			it("should show confirm location prompt when confirm location", async () => {
+				fetchAddressSpy.mockImplementation((queryString, pageNumber, onSuccess) => {
+					onSuccess(mock1PageFetchAddressResponse);
+				});
+
+				fireEvent.change(getLocationSearchInput(), { target: { value: "found somthing" } });
+
+				expect(fetchAddressSpy).toHaveBeenCalled();
+
+				await waitFor(() => {
+					expect(getLocationSearchResults(true)).not.toBeEmptyDOMElement();
+				});
+
+				const resultContainer = getLocationSearchResults();
+				const selectedResult = resultContainer.getElementsByTagName("div")[0];
+				fireEvent.click(selectedResult);
+
+				await waitFor(() => {
+					expect(selectedResult).toHaveAttribute("data-testid", expect.stringContaining("active"));
+					expect(getLocationModalControlButtons("Confirm")).not.toHaveAttribute("disabled");
+				});
+
+				fireEvent.click(getLocationModalControlButtons("Confirm"));
+
+				await waitFor(() => {
+					expect(confirmLocationOnClickSpy).toBeCalled();
+					expect(getConfirmLocationModal(true)).toBeInTheDocument();
+				});
+				await new Promise((resolve) => setTimeout(resolve, 3500));
```

Comment:
instead of waiting for 3.5s, you should mock the timers via `jest.useFakeTimers();`
it comes with functions to mock other aspects like `jest.advanceTimersByTime(3500);`

for more info: https://jestjs.io/docs/timer-mocks